pyo3-macros-backend/src/pymethod.rs

```diff
@@ -1024,17 +1029,22 @@ fn handle_error(
 
 fn extract_object(
     extract_error_mode: ExtractErrorMode,
+    holders: &mut Vec<TokenStream>,
     py: &syn::Ident,
     name: &str,
     source: TokenStream,
 ) -> TokenStream {
+    let holder = syn::Ident::new(&format!("holder_{}", holders.len()), Span::call_site());
+    holders.push(
+        quote! { let mut #holder = _pyo3::impl_::extract_argument::FunctionArgumentHolder::INIT; },
+    );
     handle_error(
         extract_error_mode,
         py,
         quote! {
             _pyo3::impl_::extract_argument::extract_argument(
                 #source,
-                &mut { _pyo3::impl_::extract_argument::FunctionArgumentHolder::INIT },
```

Comment:
I could not produce a test case actually triggering the issue but changed it for completeness. I did put it into a separate commit though if it turns out that we do not need this after all.